src/analyze/analyze-time-data.c

```diff
@@ -81,7 +82,10 @@ int acquire_boot_times(sd_bus *bus, bool require_finished, BootTimes **ret) {
         if (require_finished && times.finish_time <= 0)
                 return log_not_finished(times.finish_time);
 
-        if (arg_runtime_scope == RUNTIME_SCOPE_SYSTEM && times.security_start_time > 0) {
+        if (arg_runtime_scope == RUNTIME_SCOPE_SYSTEM && times.softreboot_start_time > 0)
+                /* On soft-reboot ignore kernel/firmware/initrd times as they are from the previous boot */
+                times.firmware_time = times.loader_time = times.kernel_time = times.initrd_time = 0;
```

Comment:
Can I confirm that this means that after soft-reboot `systemd-analyze plot` provides something different from before the soft-reboot?

Reply:
Hmm, the behaviour for scope, slice, and device units is not great. We plot shows the units as stretching from the original boot until after the reboot. See attached image.

![plog](https://github.com/systemd/systemd/assets/349618/578570ae-059b-4ebe-8b6c-afb762ae0d5d)

Reply:
Yeah I hadn't checked 'plot', I'll see to fix that

Reply:
It doesn't load correctly from github for some reason. A copy here: https://fedorapeople.org/~zbyszek/plog.svg

Reply:
But as far as I remember, `systemctl plot` had problems after systemd was reexecuted already before. This PR fixes some systemd-analyze and a bunch of other things, so I think we should merge it. `systemctl plot` is fringe functionality.

Reply:
+1 for merging this as-is. We can deal with plot separately later.

Reply:
ok, will send a follow-up

Reply:
https://github.com/systemd/systemd/pull/31988